### PR TITLE
Add oxygen alert to idle notifier

### PR DESCRIPTION
**Pull request by robinwithes**
Added a new plugin that can alert the player in different ways when low on oxygen,
the player can configure the oxygen threshold and types of warnings himself.
Fixes https://github.com/runelite/runelite/issues/4890

### CODE REVIEW

**Comment by deathbeam** [comment]
Should this really be separate plugin? Imo it would be fine if it would be in Idle Notifier. Also, the configuration for the plugin is really excessive I would expect just threshold and thats it (and threshold at 0 means disabled)

**Comment by robinwithes** [comment]
> Should this really be separate plugin? Imo it would be fine if it would be in Idle Notifier. Also, the configuration for the plugin is really excessive I would expect just threshold and thats it (and threshold at 0 means disabled)

I was looking for plugins that could fit this and could'n find any, idle notifier is a good choice though and if you want me to i can integrate this into it.

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierConfig.java [hunk 1]
```diff
@@ -96,4 +96,34 @@ default int getPrayerThreshold()
...
+		position = 8,
+		description = "Get a notifier even when not in runelite when your oxygen is about to run out")
+	default boolean oxygenNotifyAlert()
+	{
+		return true;
+	}
+
+	@ConfigItem(
+		keyName = "oxygenThreshold",
+		name = "Oxygen Threshold",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierConfig.java]: Follow naming convention of prayer/hp threshold etc. E.g, this should be
```
@ConfigItem(
  keyName = "oxygen",
  name = "Oxygen Notification Threshold",
  decription = "The amount of remaining oxygen to send a notification at. A value of 0 will disable notification."
)
default int getOxygenThreshold()
{
  return 0;
}
```

Remove rest of config for oxygen alerter except this

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 2]
```diff
@@ -531,4 +537,21 @@ private void resetTimers()
 			lastInteract = null;
 		}
 	}
+
+	@Subscribe
+	private void onVarbitChanged(VarbitChanged ev)
+	{
+		oxygenThreshold = config.oxygenAlertAmount();
+		if (oxygenThreshold >= 100 || oxygenThreshold < 0)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: remove this entire if

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 3]
```diff
@@ -531,4 +537,21 @@ private void resetTimers()
...
+		{
+			oxygenThreshold = 20;
+		}
+		if (client.getVar(Varbits.OXYGEN_LEVEL) <= oxygenThreshold * 10 && client.getVar(Varbits.OXYGEN_LEVEL) > 0)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: add check for oxygenThreshold being > 0

Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: change all these ifs to be early returns to make the method easier to read

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 4]
```diff
@@ -531,4 +537,21 @@ private void resetTimers()
...
+		{
+			if (oxygenThreshold == client.getVar(Varbits.OXYGEN_LEVEL) * 0.1)
+			{
+				notifier.notify("Your oxygen is dangerously low! Currently: " + client.getVar(Varbits.OXYGEN_LEVEL) * 0.1 + "%");
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: Maybe use same format as the HP and Prayer notifier use

Comment (robinwithes) [reply runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: I modified it to be close to the hp one.

File: runelite-api/src/main/java/net/runelite/api/Varbits.java [hunk 5]
```diff
@@ -408,6 +408,10 @@
 	 */
 	ACCOUNT_TYPE(1777),
 
+	/**
+	 * The varbit that stores the oxygen percentage for player
+	 */
+	OXYGEN_LEVEL(5811),
```
Comment (NathenSample) [inline runelite-api/src/main/java/net/runelite/api/Varbits.java]: new line after this

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 6]
```diff
@@ -123,11 +124,11 @@ public void onAnimationChanged(AnimationChanged event)
 			case WOODCUTTING_DRAGON:
 			case WOODCUTTING_INFERNAL:
 			case WOODCUTTING_3A_AXE:
-			/* Cooking(Fire, Range) */
+				/* Cooking(Fire, Range) */
```
Comment (NathenSample) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: What's going on with the formatting here?

**Pull request merged**